src/ofxGuiJson.cpp

```diff
@@ -0,0 +1,250 @@
+        return nullptr; // Return nullptr explicitly if not found or null
+    }
+}
+
+
+void ofxGuiJson::updateGuiLabels(const std::map<std::string, std::string>& labelUpdates) { // potentially only have string string instead of map
+    for (const auto& update : labelUpdates) {
+        auto labelElement = std::dynamic_pointer_cast<ofxLabel>(getGuiElement(update.first));
+        if (labelElement) {
+            labelElement->setName(update.second);
+        }
+        else {
+            ofLogError() << "Label with ID '" << update.first << "' not found for update.";
+        }
+    }
+}
+
+void ofxGuiJson::createAndAddToggle(const std::string& label, const std::string& baseId) {
+    static int toggleCount = 0; // Static counter to ensure unique IDs
+    std::string id = baseId + std::to_string(toggleCount++); // Generate unique ID
+
+    auto toggle = std::make_shared<ofxToggle>();
+    float width = 100; // Default width
+    float height = 20; // Default height
+    float posX = 50;   // Default x position
+    float posY = 50 + (toggleCount * (height + 10)); // Stack toggles vertically
+
+    toggle->setup(label, false, width, height);
+    toggle->setPosition(posX, posY);
+    gui.add(toggle.get());
+    guiElementsMap[id] = toggle;
+    dynamicToggles.insert(id); // Maintain a set of dynamically added toggles
+
+    // Adjust GUI size or enable scrolling as needed
+    if (posY + height > gui.getHeight()) {
+        gui.setSize(gui.getWidth(), posY + height + 20);
+    }
+
+    ofLogNotice() << "New toggle added: " << label << " with ID: " << id;
+}
+
+
+void ofxGuiJson::removeLastToggle() {
+    if (!guiElementsMap.empty()) {
+        auto it = guiElementsMap.end();
+        do {
+            if (it == guiElementsMap.begin()) {
+                ofLogError() << "No dynamic toggles to remove.";
+                return;  // Exit if there are no dynamic toggles or if we've checked all toggles
+            }
+            --it;  // Move to the previous element
+        } while (dynamicToggles.find(it->first) == dynamicToggles.end());  // Continue while the toggle is not dynamic
+
+        // Remove the toggle from the dynamicToggles set and guiElementsMap
+        dynamicToggles.erase(it->first);
+        guiElementsMap.erase(it);
+
+        // Rebuild the GUI
+        rebuildGui();
+        ofLogNotice() << "Dynamic toggle removed.";
+    }
+    else {
+        ofLogNotice() << "No toggles to remove.";
+    }
+}
+
+void ofxGuiJson::rebuildGui() {
+    gui.clear();
+    for (const auto& id : staticToggles) { 
+        if (guiElementsMap.find(id) != guiElementsMap.end()) {
+            gui.add(guiElementsMap[id].get());
+        }
+    }
+    for (const auto& id : dynamicToggles) {  // Ensure dynamic elements are added after static ones.
+        if (guiElementsMap.find(id) != guiElementsMap.end()) {
+            gui.add(guiElementsMap[id].get());
+        }
+    }
+}
+
+void ofxGuiJson::setBackgroundColor() {
```

Comment:
Should setting the bgColor be a part of this class?
I just ask because all we do is call `ofSetBackgroundColor`. It has nothing else to do with this class, except being able to read the color from a file. that is nice, but then it locks users to specifying bg color in the config file.

Reply:
i think it is appropriate for this to be a part of the class as it aligns with the current functionality of allowing the user to customize the GUI through a JSON